### ISSUE TITLE
Newtonsoft.Json.JsonReaderException: Input string '2.0' is not a valid integer in my application after Export/Import

### ISSUE THREAD

**Issue by hinault** [AzureCosmosDB/data-migration-desktop-tool#77]
Hi everyone

I am using DMT to export data from Cosmos DB to JSON files. Then, these data are re-imported from JSON to Cosmos DB.

The export/import process is successful. However, I am getting the following exception in my application, even though everything was working correctly before the export/import:

> Newtonsoft.Json.JsonReaderException: Input string '2.0' is not a valid integer. Path '[0].version', line 1, position 186.
   at Newtonsoft.Json.JsonTextReader.ParseReadNumber(ReadType readType, Char firstChar, Int32 initialPosition)
   at Newtonsoft.Json.JsonTextReader.ReadNumberValue(ReadType readType)
   at Newtonsoft.Json.JsonTextReader.ReadAsInt32()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateList(IList list, JsonReader reader, JsonArrayContract contract, JsonProperty containerProperty, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at Microsoft.Azure.Cosmos.CosmosJsonDotNetSerializer.FromStream[T](Stream stream)
   at Microsoft.Azure.Cosmos.CosmosJsonSerializerWrapper.FromStream[T](Stream stream)
   at Microsoft.Azure.Cosmos.CosmosJsonSerializerWrapper.FromStream[T](Stream stream)
   at Microsoft.Azure.Cosmos.CosmosSerializerCore.FromFeedStream[T](Stream stream)
   at Microsoft.Azure.Cosmos.Serializer.CosmosElementSerializer.GetResourcesHelper[T](IReadOnlyList`1 cosmosElements, CosmosSerializerCore serializerCore, CosmosSerializationFormatOptions cosmosSerializationOptions)
   at Microsoft.Azure.Cosmos.Serializer.CosmosElementSerializer.GetResources[T](IReadOnlyList`1 cosmosArray, CosmosSerializerCore serializerCore)
   at Microsoft.Azure.Cosmos.QueryResponse`1..ctor(HttpStatusCode httpStatusCode, IReadOnlyList`1 cosmosElements, CosmosQueryResponseMessageHeaders responseMessageHeaders, CosmosDiagnostics diagnostics, CosmosSerializerCore serializerCore, CosmosSerializationFormatOptions serializationOptions, RequestMessage requestMessage)
   at Microsoft.Azure.Cosmos.QueryResponse`1.CreateResponse[TInput](QueryResponse cosmosQueryResponse, CosmosSerializerCore serializerCore)
   at Microsoft.Azure.Cosmos.CosmosResponseFactoryCore.CreateQueryFeedResponseHelper[T](ResponseMessage cosmosResponseMessage)
   at Microsoft.Azure.Cosmos.CosmosResponseFactoryCore.CreateQueryFeedUserTypeResponse[T](ResponseMessage responseMessage)
   at Microsoft.Azure.Cosmos.FeedIteratorCore`1.ReadNextAsync(ITrace trace, CancellationToken cancellationToken)
   at Microsoft.Azure.Cosmos.ClientContextCore.RunWithDiagnosticsHelperAsync[TResult](String containerName, String databaseName, OperationType operationType, ITrace trace, Func`2 task, Func`2 openTelemetry, String operationName, RequestOptions requestOptions)
   at Microsoft.Azure.Cosmos.ClientContextCore.OperationHelperWithRootTraceAsync[TResult](String operationName, String containerName, String databaseName, OperationType operationType, RequestOptions requestOptions, Func`2 task, Func`2 openTelemetry, TraceComponent traceComponent, TraceLevel traceLevel)
   at Prodago.Common.Infrastructure.Cosmos.CosmosHelper.ToListAsync[T](IQueryable`1 cosmosLinqQuery, Boolean useFeedIterator, CancellationToken cancellationToken) in C:\Users\hdonfack\source\repos\Prodago2\backend\Source\Prodago.Common.Infrastructure\Cosmos\CosmosHelper.cs:line 0
   at Prodago.Common.Infrastructure.Cosmos.CosmosService.GetEntitiesInContainer[T,TResult](String containerId, String partitionKey, Func`2 linqApplier, CancellationToken cancellationToken, Boolean allowCaching) in C:\Users\hdonfack\source\repos\Prodago2\backend\Source\Prodago.Common.Infrastructure\Cosmos\CosmosService.cs:line 138
   at Prodago.Common.Infrastructure.Cosmos.CosmosService.GetEntitiesInContainer[T](String containerId, String partitionKey, Func`2 linqApplier, CancellationToken cancellationToken, Boolean allowCaching) in C:\Users\hdonfack\source\repos\Prodago2\backend\Source\Prodago.Common.Infrastructure\Cosmos\CosmosService.cs:line 142
   at XXX.Repositories.TenantRegistryVersionRepository.DisableCurrentVersion() in TenantRegistryVersionRepository.cs:line 95

The exception occurs on the 'version' field of the document : 

```json
{
    "tenantName": "xxx",
    "registryFilename": "xxxx01202305302023063004.0",
    "isCurrent": true,
    "creationDate": "2023-06-30T18:15:25.2346819+00:00",
    "version": 2,
    "id": "ddd44d1f-cb5a-4172-b36d-436e4b5e8fd6",
    "_rid": "QPpiAKZdqj0BAAAAAAAAAA==",
    "_self": "dbs/QPpiAA==/colls/QPpiAKZdqj0=/docs/QPpiAKZdqj0BAAAAAAAAAA==/",
    "_etag": "\"00005713-0000-0a00-0000-64a41f5d0000\"",
    "_attachments": "attachments/",
    "_ts": 1688477533
}
```

Do you have any idea why this is happening and how I could work around it?

Thanks

**Comment by hinault** [comment]
I pretended to modify and save the item in the data explorer in the azure portal, and I no longer got the exception in my application.

**Comment by bowencode** [comment]
Duplicate of #71 , fixed in Release 2.1.2
